From: Hiral Shah <hishah@cisco.com>
Date: Thu, 20 Feb 2014 15:44:01 -0800
Subject: fnic: Sending single FIP solicitation frame in non-FIP mode
Patch-Mainline: Not yet
References: bnc#863365,FATE#315811

During OS initialization fnic was sending a single FIP solicitation
which it should not be doing. Explicitly setting fcoe control structure
state flag to FIP_ST_NON_FIP to avoid sending FIP frame in case of
non-FIP mode.

Signed-off-by: Hiral Shah <hishah@cisco.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/fnic/fnic_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/scsi/fnic/fnic_main.c b/drivers/scsi/fnic/fnic_main.c
index 33e4ec2..09cde8d 100644
--- a/drivers/scsi/fnic/fnic_main.c
+++ b/drivers/scsi/fnic/fnic_main.c
@@ -773,6 +773,7 @@ static int fnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		shost_printk(KERN_INFO, fnic->lport->host,
 			     "firmware uses non-FIP mode\n");
 		fcoe_ctlr_init(&fnic->ctlr, FIP_MODE_NON_FIP);
+		fnic->ctlr.state = FIP_ST_NON_FIP;
 	}
 	fnic->state = FNIC_IN_FC_MODE;
 
-- 
1.7.12.4

